cg/server/admin.py

```diff
@@ -168,6 +191,27 @@ def view_application_link(unused1, unused2, model, unused3):
             else ""
         )
 
+    def on_model_change(self, form: Form, model: Application, is_created: bool):
+        """Override to persist entries to the OrderTypeApplication table"""
+        super(ApplicationView, self).on_model_change(form=form, model=model, is_created=is_created)
+        order_types: list[OrderType] = form["suitable_order_types"].data
+        db.delete_order_type_applications_by_application_id(model.id)
+        db.session.add_all(
+            db.link_order_types_to_application(application=model, order_types=order_types)
+        )
+        db.session.commit()
```

Comment:
I would say that we either keep these here or we move them to the Application service introduced by @diitaz93 in #3864.

Reply:
Maybe wrap the db operations inside a try-except to avoid pending transactions. 

There is another method in the CaseView (`set_action_for_cases`) which handles a session and commits. Not a fan either but I can live with that. There is also some delivered message in this function to the user, maybe something to consider??

Reply:
And I think that there is no need to involve the ApplicationsWebService

Reply:
Getting conflicting advice here. Unless @Vince-janv is also strongly against, I will move it to the application service. I think there is a bigger issue with having the database persisting in the admin views which are already complicated enough than the scope of the service changing

Reply:
I'm for moving 🤚